inspirehep/modules/disambiguation/records.py

```diff
@@ -0,0 +1,162 @@
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Set of methods to create and update records."""
+
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+)
+
+from celery.utils.log import get_task_logger
+from flask import current_app, url_for
+
+from invenio_db import db
+
+from invenio_indexer.api import RecordIndexer
+
+from invenio_records import Record
+from invenio_records.signals import after_record_insert
+
+from inspirehep.dojson import utils as inspire_dojson_utils
+from inspirehep.modules.pidstore.minters import inspire_recid_minter
+
+from inspirehep.modules.disambiguation.receivers import (
+    append_new_record_to_queue,
+)
+
+logger = get_task_logger(__name__)
+
+
+def _get_author_schema():
+    """Generate schema URL to be inserted in $schema field."""
+    # FIXME: Use a dedicated method when #1355 will be resolved.
+    return url_for(
+        "invenio_jsonschemas.get_schema",
+        schema_path=current_app.config.get(
+            'DISAMBIGUATION_AUTHORS_SCHEMA_PATH'
+        )
+    )
+
+
+def create_author(profile):
+    """Create a new author profile based on a given signature.
+
+    The method receives a dictionary representing an author.
+    Based on the values, it creates a dictionary in the invenio_records format.
+    After all the fields are processed, the method calls create_record
+    from invenio_records.api to put the new record.
+
+    :param profile:
+        A signature representing an author's to be created as a profile
+        with additional values for an experiment and a field_categories
+        taken from the hep paper.
+
+        Example:
+            profile = {u'affiliations': [{u'value': u'Yerevan Phys. Inst.'}],
+                       u'alternative_name': None,
+                       u'curated_relation': False,
+                       u'email': None,
+                       u'full_name': u'Chatrchyan, Serguei',
+                       u'inspire_id': None,
+                       u'orcid': None,
+                       u'profile': u'',
+                       u'recid': None,
+                       u'role': None,
+                       u'uuid': u'd63537a8-1df4-4436-b5ed-224da5b5028c'}
+
+    :return:
+        A recid, where the new profile can be accessed.
+
+        Example:
+            "1234"
+    """
+    # Get the author's name.
```

Comment:
This style of commenting is only appropriate when writing a tutorial for people that have never programmed before. It's completely obvious that `name = profile.get('full_name')` gets the author's name; if it didn't, something would be very very wrong. Comments should never say _what_ are you doing, but _why_ are you doing it.